projects/dsp-ui/src/lib/viewer/services/event-bus.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+@Injectable({
```

Comment:
Could you add some information how to use this service, especially when to subscribe and unsubscribe to it? Maybe the design docs would be a good place for that.

Since the service is provided in root), there is only a single instance (singleton). This means that it can not be used to perform two actions in parallel. How can this be ensured?

Reply:
> Since the service is provided in root, there is only a single instance (singleton). This means that it can not be used to perform two actions in parallel. How can this be ensured?

You raise a good question 🤔

Reply:
It is possible to provide a service at the component level: https://angular.io/guide/providers#limiting-provider-scope-with-components.

However, I don't know how to subscribe to this very instance then.

Reply:
I'll wait to write the docs until we settle on the correct implementation. I've pushed some changes to provide the service at the component level in 80b694b